addons/sale_mrp/models/account_move.py

```diff
@@ -11,8 +11,10 @@ def _stock_account_get_anglo_saxon_price_unit(self):
 
         so_line = self.sale_line_ids and self.sale_line_ids[-1] or False
         if so_line:
-            bom = so_line.product_id.product_tmpl_id.bom_ids and so_line.product_id.product_tmpl_id.bom_ids[0]
-            if bom.type == 'phantom':
+            boms = self.sale_line_ids.move_ids.bom_line_id.bom_id
+            bom = boms.filtered(lambda bom: bom.type == "phantom" and bom.product_id == so_line.product_id
```

Comment:
IMO, there is a mistake here the `and` is apply first it means that you accept the bom if `bom.type != "phantom"` and `not bom.product_id and bom.product_tmpl_id == so_line.product_template_id` is fulfilled (which is wrong).
Suggestion: 
```python
boms.filtered(lambda bom: bom.type == "phantom" and (bom.product_id == so_line.product_id
                                            or (not bom.product_id and bom.product_tmpl_id == so_line.product_template_id)))
```
or used a intern method `filter_kit_of_product` to be more readable.